test/files.js

```diff
@@ -299,7 +306,10 @@ describe('files', function () {
       )
     })
 
-    it('trickle DAGs', () => {
+    // Somehow, the hashes for the same content are not equal
+    // This problem was also detected on
+    // both 'remove a child shared by multiple pins' test and 'print same pins'
+    it.skip('trickle DAGs', () => {
```

Comment:
Does this still need to be skipped?

Reply:
These 3 tests cannot be unskipped as they are still failing. Issue number #88 has further details about this problem.
Should we block this PR until we get this solved?

Reply:
I'm taking a look at those failing tests

Reply:
Though having said that I do seem to be getting different CIDs from js-IPFS master on the cli if the daemon is on or off. 😬

Reply:
`ipfs-http-client` wasn't inspecting the right field for the `--trickle` arg - the fix is in https://github.com/ipfs/js-ipfs-http-client/pull/1129 and I'll backport it to `ipfs-http-client@38.0.x` when it's merged.

Reply:
Urgh, no, it's more complicated that that, sorry - tired.  `ipfs-http-client` is doing the right thing - it takes a `trickle` key as part of it's options and sends that over http.  The cli in `ipfs` takes a `trickle` argument but turns it into `strategy: 'trickle'` (as that's what's required by the `unixfs-importer`) and passes that to `ipfs.add`* and the http api endpoint does the same thing.

The right thing to do is for `ipfs-http-client`,  the cli and the http api endpoint to all pass `trickle: true` to `ipfs.add` (as per the interface spec) and have *that* do the conversion to `strategy: 'trickle'`.

* = actually `ipfs._addAsyncIterator` but `ipfs.add` just passes it's options straight through so same same.

Reply:
https://github.com/ipfs/js-ipfs/pull/2541